File: problema 10/c-languaje/problema_10.c
#include <stdio.h>
#include <string.h>

void ordenarPorPromedio(char nombres[][50], float promedios[], int N) {
    int i, j;
    for (i = 0; i < N - 1; i++) {
        for (j = 0; j < N - 1 - i; j++) {
            if (promedios[j] < promedios[j + 1]) {
                float tempPromedio = promedios[j];
                promedios[j] = promedios[j + 1];
                promedios[j + 1] = tempPromedio;
                
                char tempNombre[50];
                strcpy(tempNombre, nombres[j]);
                strcpy(nombres[j], nombres[j + 1]);
                strcpy(nombres[j + 1], tempNombre);
            }
        }
    }
}

int main() {
    int N, i;
    printf("Ingrese el número de alumnos: ");
    scanf("%d", &N);
    char nombres[N][50];
    float promedios[N];

    for (i = 0; i < N; i++) {
        printf("Ingrese el nombre del alumno %d: ", i + 1);
        scanf("%s", nombres[i]);
        printf("Ingrese el promedio de %s: ", nombres[i]);
        scanf("%f", &promedios[i]);
    }

    ordenarPorPromedio(nombres, promedios, N);

    printf("\nAlumnos ordenados por promedio:\n");
    for (i = 0; i < N; i++) {
        printf("%s: %.2f\n", nombres[i], promedios[i]);
    }

    return 0;
}

File: problema 10/fortran/problema_10.f90
program ordenarAlumnos
    implicit none
    integer :: N, i, j
    character(len=50), allocatable :: nombres(:)
    real, allocatable :: promedios(:)
    real :: tempPromedio
    character(len=50) :: tempNombre

    print *, 'Ingrese el número de alumnos:'
    read *, N

    allocate(nombres(N))
    allocate(promedios(N))

    do i = 1, N
        print *, 'Ingrese el nombre del alumno ', i, ':'
        read *, nombres(i)
        print *, 'Ingrese el promedio de ', nombres(i), ':'
        read *, promedios(i)
    end do

    ! Ordenar por promedio
    do i = 1, N-1
        do j = 1, N-i
            if (promedios(j) < promedios(j+1)) then
                tempPromedio = promedios(j)
                promedios(j) = promedios(j+1)
                promedios(j+1) = tempPromedio

                tempNombre = nombres(j)
                nombres(j) = nombres(j+1)
                nombres(j+1) = tempNombre
            end if
        end do
    end do

    print *, 'Alumnos ordenados por promedio:'
    do i = 1, N
        print *, nombres(i), ':', promedios(i)
    end do
end program ordenarAlumnos